### ISSUE TITLE
Automatically mark unread notification as read when corresponding page is visited

### ISSUE THREAD

**Issue by badshah400** [openSUSE/open-build-service#17222]
**Is your feature request related to a problem? Please describe.**
It is quite a frustrating experience to go back and mark each unread notification as read after one has already taken action on the associated request or comment. For example, if I click on an unread notification for a review on an sr and ack that request, the "Mark as read" button at the top right corner of the page disappears. I have to go back to the list of unread notifications, find the notification corresponding to the request I just acked and mark it as read manually.

**Describe the solution you'd like**
Automatically mark the notification as read if one clicks on it and visits the corresponding sr, project, or package page. This would make it similar to how github handles unread notifications and reduce *a lot* of spurious work just pruning through the list of unread notifications.

**Comment by saraycp** [comment]
I'm not sure the described behavior is what we want. 
In other tools, once you visit the element associated to the notification and go back to the notifications list, you'll still find the notification in the list but in gray text color. The change of color represents that it was already visited but it's still there in case you haven't finished with it yet.

If we mark it as read, as suggested in this issue, it will disappear from the list. And that will happen even if you clicked by mistake or if you decided to continue handling the notification later. It will be hard to find it again.

I wouldn't go further with this approach. I opened another issue which could improve the scenario a bit: https://github.com/openSUSE/open-build-service/issues/20252

**Comment by badshah400** [comment]
> I'm not sure the described behavior is what we want. In other tools, once you visit the element associated to the notification and go back to the notifications list, you'll still find the notification in the list but in gray text color. The change of color represents that it was already visited but it's still there in case you haven't finished with it yet.

I use notifications on my (Android) phone every day and never see notifications persisting even after having intentionally tapped them. Yes, they are available from a "Notifications history" option, but that's still marked as read. Even in your description, the colour change represents a change of unread -> read status of the notification. My request is exactly the same: mark a clicked upon notification as _read_, not delete it from the universe. Not sure what the misunderstanding here is.

> 
> If we mark it as read, as suggested in this issue, it will disappear from the list. And that will happen even if you clicked by mistake or if you decided to continue handling the notification later. It will be hard to find it again.

Notifications marked as "read" can still be seen using the interface already existing today and without any trouble. Here is a screenshot to show what I mean:

<img width="1592" height="472" alt="Image" src="https://github.com/user-attachments/assets/e1d9235e-7ef7-4a6c-92a6-8935e07f4237" />

> 
> I wouldn't go further with this approach. I opened another issue which could improve the scenario a bit: [#20252](https://github.com/openSUSE/open-build-service/issues/20252)

That's a totally different issue and you seem to have misunderstood the current request.

**Comment by saraycp** [comment]
I totally agree the notifications in Android, for example, work like you describe.
My explanation was taking the example of GitHub notifications. They have the actions _Mark as Read_ that mainly changes the text color and the action _Done_ that makes it disappear from the list.

We in OBS don't have two kinds of actions. We'll need to figure out what's the behavior that better fits our needs. Let us think a bit on it.

**Comment by saraycp** [comment]
After some discussion, we've decided that this is not the behavior we want. We don't want the notification to disappear as soon as it's visited.

Thank you for you proposal. I'm closing but it will be nice to receive more from you.
